### ISSUE TITLE
libosmscout-client-qt.pro referes to nonexistent files

### ISSUE THREAD

**Issue by asanoaozora** [Framstag/libosmscout#557]
https://github.com/Framstag/libosmscout/blob/0b7c3ae833e661b4db2d2798ffce43c7362ec192/libosmscout-client-qt/libosmscout-client-qt.pro#L23
https://github.com/Framstag/libosmscout/blob/0b7c3ae833e661b4db2d2798ffce43c7362ec192/libosmscout-client-qt/libosmscout-client-qt.pro#L28
https://github.com/Framstag/libosmscout/blob/0b7c3ae833e661b4db2d2798ffce43c7362ec192/libosmscout-client-qt/libosmscout-client-qt.pro#L55
https://github.com/Framstag/libosmscout/blob/0b7c3ae833e661b4db2d2798ffce43c7362ec192/libosmscout-client-qt/libosmscout-client-qt.pro#L60

when I commented the lines it built, but not sure it's really the best way to proceed (because I don't know the content of these files)

**Comment by Karry** [comment]
Hi. How do you build it? We currently supports (actively) two build systems: CMake and Meson. This Qt project file is probably relic and should be safe to remove it... @vyskocil how do you build Qt libraries on MacOS, it is safe to remove existing *.pro files and autoconf support?

**Comment by Framstag** [comment]
I also would claim these project files as unsupported. The have not been touched for some time. They are also not tested in any of the central builds IMHO.

Currently the cmake build is the most advanced build. The meson build only supports only a subset of the features, partly because of missing features in meson itself (=> Mac OS/iOS and installation in general).

The autoconf build is also depreciated, not tested in central build either and very likely already broken and the relevant files are only not deleted because we still wait for final feedback from the Mac OS / iOS users regarding them still being required (for generation of some header files).

Thus: Please build as suggested in issue #556 If these build breaks you got us and we have to definitely help you and fix it for you :-) Also give us further information regarding your concrete build environment.

**Comment by asanoaozora** [comment]
Something is missing in libosmscout:
nm  /usr/local/lib/libosmscout.so.0.0.0 | grep RoutingDatabase | grep ' U '
                 U _ZN8osmscout15RoutingDatabase12GetJunctionsERKSt3setImSt4lessImESaImEERSt6vectorISt10shared_ptrINS_12IntersectionEESaISB_EE
                 U _ZN8osmscout15RoutingDatabase4OpenERKSt10shared_ptrINS_8DatabaseEE
                 U _ZN8osmscout15RoutingDatabase5CloseEv
                 U _ZN8osmscout15RoutingDatabaseC1Ev
Reason is that osmscout/routing/RouteDB.lo is missing.
Unfortunately I don't know how to fix it :-(
Could you please fix it ?

**Comment by vyskocil** [comment]
Karry is right, the .pro files and XCode projects are not maintained anymore in favor of the CMake build that works mostly for me. I'm still looking to make it 100% functional  for iOS and OSX...

**Comment by Framstag** [comment]
@asanoaozora That is strange. As you can see from https://github.com/Framstag/libosmscout/blob/master/libosmscout/CMakeLists.txt (search for src/osmscout/routing/RoutingDB.cpp) RoutingDB.cpp is part of the CMake file and thus RoutingDB.lo should be build (As all other files) and linked to the library. Using http://demangler.com/  in fact the missing symbols do come from the named file.

Please make sure that you the cmake file and the RouteDB.cpp a euqls to the versions at github. Please check, if RoutingDB.cpp is build during the cmake build.

I'll recheck in parallel with the Ubuntu 17.10 gcc cmake docker image in the ci/docker sub directory.

**Comment by Framstag** [comment]
The ubuntu docker builds without any errors (after I changed the build script to use the github repository). The builds does create *.cpp.o file, so if you locally see *.lo files this might hint at an libtool based autoconf build and you are still mixing cmake and autoconf builds. If you are unsure just erase the git repository directory locally and recheckout it again.

**Comment by Framstag** [comment]
@asanoaozora: Did you make progress? How can we give you further help?

**Comment by asanoaozora** [comment]
Hi, sorry for the late answer (some private issues). Your help has been very useful to understand what I did wrong. I agree that I mixed cmake and autoconf. After some clean up I rebuilt from scratch and now the build of your stuff passed successfully. I'm now working on the extension of my Qt based application. There's an example based on QML (OSMScout) but I don't use QML, so I need to rework it a little bit. 
Since I moved to Qt5.9.1 (requested by the  libosmscout_client_qt) I got some other pb to fix, as usual..
As soon as I'll have make more progress, I'll push my code to Github, but for the moment it's not sufficiently clean..
